### DISCUSSION TITLE
Avoid TypeScript features that require transformation

### DISCUSSION THREAD

**Discussion by tien** [polkadot-api/polkadot-api#619] [Ideas]
In particular, the use of `enum` is a dated feature of TypeScript that isn't part of the JavaScript standard. This means it requires code transformation rather than just stripping the types.

This has become more important with the [option](https://github.com/nodejs/node/blob/main/doc/api/typescript.md#type-stripping) to use TypeScript with Node.js, which requires avoiding TypeScript features that need transformation.

Alternatives include using string literal union or [const object](https://www.typescriptlang.org/docs/handbook/enums.html#objects-vs-enums).

**Comment by timbrinded** [comment]
Not using `enum` types should be because they are strictly inferior to string literal unions or zod enums. 

The Typescript strip types feature is very experimental right now and also a bit lacklustre, affordances shouldn't be made for it.

**Reply by tien** [reply]
**Personally** I would agree that using string literal union is generally better (not sure about Zod though), but "strictly" is a rather strong word :p

I believe staying as close to vanilla JS spec as possible (minus the types of course 🤣) is more relevant here, it just happens to be highlighted even further now with the possibility of native TS support in Node.js.
